### DISCUSSION TITLE
Add/Update/Delete nested objects in object array

### DISCUSSION THREAD

**Discussion by ben-me** [neuefische/questions#229] [Web]
Hello,

my application features a list of rooms which contain desks which can be reserved by a user.
To display this I build the array like this:

```jsx

const rooms = [
  {
    id: '1',
    name: 'Room 1',
    desks: [
      {
        id: '1',
        name: 'Desk 1',
        reservations: [
          {
            id: '1',
            starttime: '08:00',
            endtime: '10:00',
            date: '2022-09-29',
            user: 'user1',
            isPrivate: false,
          },
          ]
```

It's one big array with Room objects. Those also have a key called "desks" with an array as value. In this array are desk objects which have a key called "reservations". This is another array with reservation objects.

Quite a lot of nested object. The whole array is also loaded into a useState so I might even have to work with the setter function of said state.
Right now the Goal is to add/remove or update reservations in a specified room/desk combination.

For example: I want to delete reservation with id "438294732" in room 1, desk 1. The same applies to adding reservations or updating existing ones (finding them with their unique id)

How do I properly access and manipulate deeply nested objects?

**Comment by julialoeschel** [comment]
Hey there.
updating an very nested object is tricky - but you can do it. 
You need to combine several methods that youve learned with us, and also some new ones. 
have a closer look at e.g. Object.keys() method , splice(), indexOf() and all this kind of stuff. 

good luck ;)
